### ISSUE TITLE
0.8.x -- traceback

### ISSUE THREAD

**Issue by thezoggy** [sabnzbd/sabnzbd#318]
updated to latest, seeing this now after job finishes:
```
2015-09-07 08:05:55,986::ERROR::[_cplogging:216] [07/Sep/2015:08:05:55] HTTP 
Traceback (most recent call last):
  File "C:\Program Files (x86)\sabnzbd_0.8.x\cherrypy\_cprequest.py", line 670, in respond
    response.body = self.handler()
  File "C:\Program Files (x86)\sabnzbd_0.8.x\cherrypy\lib\encoding.py", line 217, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "C:\Program Files (x86)\sabnzbd_0.8.x\cherrypy\_cpdispatch.py", line 61, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "C:\Program Files (x86)\sabnzbd_0.8.x\sabnzbd\interface.py", line 420, in tapi
    return api_handler(kwargs)
  File "C:\Program Files (x86)\sabnzbd_0.8.x\sabnzbd\api.py", line 100, in api_handler
    response = _api_table.get(mode, (_api_undefined, 2))[0](name, output, kwargs)
  File "C:\Program Files (x86)\sabnzbd_0.8.x\sabnzbd\api.py", line 498, in _api_history
    output=output)
  File "C:\Program Files (x86)\sabnzbd_0.8.x\sabnzbd\api.py", line 1837, in build_history
    items = get_active_history(queue, items)
  File "C:\Program Files (x86)\sabnzbd_0.8.x\sabnzbd\api.py", line 1928, in get_active_history
    history = build_history_info(nzo)
  File "C:\Program Files (x86)\sabnzbd_0.8.x\sabnzbd\database.py", line 441, in build_history_info
    stages = decode_factory(nzo.unpack_info)
  File "C:\Program Files (x86)\sabnzbd_0.8.x\sabnzbd\database.py", line 520, in decode_factory
    for key in text:
RuntimeError: dictionary changed size during iteration
```

**Comment by thezoggy** [comment]
also, its passing a bad dir to the script:

notice it renamed file.. but then when it goes to pass the dir to the script it adds an extra release name as part of the path:
```
2015-09-07 08:05:55,967::DEBUG::[misc:1211] Renaming "\\?\C:\Users\zoggy\downloads\complete\TV\Anthony.Bourdain.No.Reservations.S05E17.Montana.720p.HDTV.x264-ORENJi\Anthony.Bourdain.No.Reservations.S05E17.Montana.720p.HDTV.x264-ORENJi\Anthony.Bourdain.No.Reservations.S05E17.Montana.720p.HDTV.x264-ORENJi.mkv" to "\\?\C:\Users\zoggy\downloads\complete\TV\Anthony.Bourdain.No.Reservations.S05E17.Montana.720p.HDTV.x264-ORENJi\Anthony.Bourdain.No.Reservations.S05E17.Montana.720p.HDTV.x264-ORENJi.mkv"
2015-09-07 08:05:55,973::INFO::[newsunpack:169] Running external script C:\Program Files (x86)\sickbeard_mp4_automator\SABPostProcess.py(C:\Users\zoggy\downloads\complete\TV\Anthony.Bourdain.No.Reservations.S05E17.Montana.720p.HDTV.x264-ORENJi\Anthony.Bourdain.No.Reservations.S05E17.Montana.720p.HDTV.x264-ORENJi\.., Anthony.Bourdain.No.Reservations.S05E17.Montana.720p.HDTV.x264-ORENJi.nzb, Anthony.Bourdain.No.Reservations.S05E17.Montana.720p.HDTV.x264-ORENJi, , bypass, alt.binaries.misc, 0, )
```
deff looks like that short path is the issue

**Comment by shypike** [comment]
From which commit did you update?

**Comment by thezoggy** [comment]
im on latest `bde88cba7a9e75ac9b7f66e306417a50a7adb457`

**Comment by shypike** [comment]
I can confirm the incorrect path passed to the script. For some odd reason it only happens on WIndows. Once I find out why, it's likely an easy fix,
I cannot reproduce the traceback at all.

**Comment by shypike** [comment]
OK, error in the Windows Python library: os.path.normpath() fails to convert "\\\\?\\d:\\tv\\bla\\bla\\.." to "\\\\?\\d:\\tv\\bla". It works fine for old-style paths. The conversion needs os.path.abspath() too.
Still no sign of the traceback.

**Comment by shypike** [comment]
Parameter part fixed in commit 7bb3dd39d42592e68aa4c136f9823ee390666577

**Comment by shypike** [comment]
Does the problem still occur when using Beta1 (or the develop branch)?
